### ISSUE TITLE
SerialUSB cause memory leak after standby, fix from main branch causes serial to stop responding

### ISSUE THREAD

**Issue by tplooker** [adafruit/ArduinoCore-samd#46]
Hi,

The crux of this issue is 

`USBDevice.detach();
LowPower.standby();
USBDevice.attach();`

Causes a memory leak of 592 bytes, this is caused by the following lines (starts at 462) in USBCore.cpp

Note - Im using an external interrupt to wake up the device which isnt shown above.

`else if (config == (USB_ENDPOINT_TYPE_BULK | USB_ENDPOINT_OUT(0)))
{
	epHandlers[ep] = new DoubleBufferedEPOutHandler(usbd, ep, 256);
}`

Whereby if the handler is already initialized, it overwrites.

This issue here https://github.com/arduino/ArduinoCore-samd/issues/293 on the main arduino SAMD branch has now been resolved, however when I make the change shown below in the USBCore.cpp

`else if (config == (USB_ENDPOINT_TYPE_BULK | USB_ENDPOINT_OUT(0)))
{
	if (epHandlers[ep] == NULL) {
                   epHandlers[ep] = new DoubleBufferedEPOutHandler(usbd, ep, 256);
        }
}`

And wake the device up from sleep, the device doesnt recieve any new information via Serial.

**Comment by ladyada** [comment]
we've merged with upstream - so this should be fixed if they fixed it in arduino's samd! will close, re-open if ya find it didn't merge right :)
